src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -51,14 +51,30 @@ inline ql::span<const ValueId> getIdsFromVariable(
 CPP_template(typename T, typename Transformation = std::identity)(
     requires SingleExpressionResult<T> CPP_and isConstantResult<T> CPP_and
         ranges::invocable<Transformation, T>)
-    cppcoro::generator<const std::decay_t<std::invoke_result_t<
+    ad_utility::InputRangeTypeErased<std::decay_t<std::invoke_result_t<
         Transformation, T>>> resultGenerator(T constant, size_t numItems,
                                              Transformation transformation =
                                                  {}) {
-  auto transformed = transformation(constant);
-  for (size_t i = 0; i < numItems; ++i) {
-    co_yield transformed;
-  }
```

Comment:
basically `views::repeat` (but not yet present in  `std::ranges`, so we have to make this work with `ql::ranges`.